backend/api/routes/newsletter.py

```diff
@@ -0,0 +1,59 @@
+from fastapi import APIRouter, Depends
+
+from starlette.requests import Request
+from sqlalchemy.orm import Session
+
+from api.crud.base import get_db
+from api.scripts.email import send_email
+from api.models.models import NewsLetter
+
+router = APIRouter()
+
+
+# subscribed user
+@router.post('/newsletter-subscribe', tags=['EmailSubscription'])
+async def newsletter_subscribed(request: Request, user_email: str, db: Session=Depends(get_db)):
+    if request:
+        users = db.query(NewsLetter).filter(NewsLetter.email == user_email).first()
+        if users is None:
+            insert_news_sub = NewsLetter(email=user_email, subscribed=True)
+            db.add(insert_news_sub)
+            db.commit()
+            db.refresh(insert_news_sub)
+            return {"isnewLetterSubscribed": {"email": insert_news_sub.email, "subscribed": insert_news_sub.subscribed}}
+        else:
+            users.subscribed = True
+            db.add(users)
+            db.flush()
+            db.commit()
+            db.refresh(users)
+            return {"isnewLetterSubscribed": {"email": users.email, "subscribed": users.subscribed}}
+        
+
+# cancel newletter subscription
+@router.post('/newsletter-unsubscribe', tags=['EmailSubscription'])
```

Comment:
**newsletter-subscription** is also the better name here but now it should be a DELETE endpoint